lib/app/dispatcher.rb

```diff
@@ -18,62 +18,82 @@ class Dispatcher
     POLL_IDLING = (App::Config[:idle_timeout] || 60).to_i
     TERMINATION_TIMEOUT = (App::Config[:termination_timeout] || 60).to_i
 
-    attr_reader :scheduler
+    @running = Concurrent::AtomicBoolean.new(false)
 
-    def initialize
-      @pool = Concurrent::ThreadPoolExecutor.new(
-        min_threads: 5,
-        max_threads: 10,
-        max_queue: 100,
-        fallback_policy: :abort
-      )
-      @is_shutting_down = false
-      @heartbeats = {}
-    end
+    class << self
+      def start!
+        running!
+        Utility::Logger.info('Starting connector service...')
+        start_polling_jobs!
+      end
 
-    def start!
-      @scheduler ||= Core::NativeScheduler.new(POLL_IDLING)
-      @scheduler.when_triggered do |connector_settings|
-        service_type = connector_settings.service_type
-        connector_id = connector_settings.id
+      def shutdown!
+        Utility::Logger.info("Shutting down connector service with pool [#{pool.class}] and scheduler [#{scheduler.class}]...")
+        running.make_false
+        scheduler.shutdown
+        pool.shutdown
+        pool.wait_for_termination(TERMINATION_TIMEOUT)
+      end
 
-        # connector-level checks
-        Core::ElasticConnectorActions.ensure_content_index_exists(connector_settings.index_name)
-        raise "[#{service_type}] is not a supported connector" unless Connectors::REGISTRY.registered?(service_type)
+      private
 
-        @pool.post do
-          send_heartbeat(connector_id, service_type)
-          Utility::Logger.info("Starting a job for [#{service_type} - #{connector_id}]...")
-          job_runner = Core::SyncJobRunner.new(connector_settings, service_type)
-          job_runner.execute
-        rescue StandardError => e
-          Utility::ExceptionTracking.log_exception(e, "Job for [#{service_type} - #{connector_id}] failed due to unexpected error.")
-        end
+      attr_reader :running
+
+      def running!
+        raise 'connector service is already running!' unless running.make_true
       end
-    rescue StandardError => e
-      Utility::ExceptionTracking.log_exception(e, 'Dispatcher failed due to unexpected error.')
-    end
 
-    def shutdown
-      Utility::Logger.info("Shutting down dispatcher with pool [#{@pool&.class}] and scheduler [#{@scheduler&.class}]...")
-      @is_shutting_down = true
-      @scheduler&.shutdown
-      @pool.shutdown
-      @pool.wait_for_termination(TERMINATION_TIMEOUT)
-    end
+      def pool
+        @pool ||= Concurrent::ThreadPoolExecutor.new(
+          min_threads: 5,
```

Comment:
could you set the default to min_thread: 0 max: 5 and surface the options in the config file (can be a follow up bug)